unittests/test_pipeline.py

```diff
@@ -542,15 +535,13 @@ def __init__(self):
                     r'Hello, World\!', self.stdout)
 
             @rfm.run_before('run')
-            def setflags(self):
-                self.post_run = ['echo hello > greetings.txt']
+            def set_post_run(self):
+                self.sourcesdir = None
 
             @rfm.run_after('run')
-            def check_executable(self):
-                outfile = os.path.join(self.stagedir, 'greetings.txt')
-
-                # Make sure that this hook is executed after wait()
-                assert os.path.exists(outfile)
+            def check_greetings(self):
+                # Make sure nothing was copied to the stage directory
+                assert len(os.listdir(self.stagedir)) == 3
```

Comment:
This test is not so robust. It will fail if the number of files generated by reframe increases. This should run before the `run` phase and you should make sure that the stage directory is empty.

Reply:
So I replaced the `unset_sourcesdir` hook with the check that the stage directory is empty before run. This was failing before the change.